### ISSUE TITLE
About the training of RL

### ISSUE THREAD

**Issue by DFLyan** [NVlabs/alpamayo#110]
When I was conducting RL training, I found that Cosmos RL uses VLLM for training, and the current code trajectory is obtained through inverse normalization instead of expert models. However, VLLM is unable to obtain KV cache for expert model trajectory generation, which seems inconsistent with the description in the paper. May I ask, if the complete version of the code cannot be used for RL training without VLLM, how can we use expert models to generate trajectories and perform reward calculations?

**Comment by super-anova** [comment]
Hi. Our released RL post training code using Cosmos RL and vLLM is currently for the VLM only. For reference: What exactly is being RL post-trained? ([ref to faq](https://github.com/NVlabs/alpamayo/tree/main/finetune/rl#faq)) (btw this page is superseded by https://github.com/NVlabs/alpamayo-recipes/tree/main/recipes/alpamayo1_x_rl#faq going forward)

In terms of the training recipe, you may want to consider something like this when post trainning the Alpamayo models with your own dataset. Here I use Alpamayo 1.5 model (https://github.com/NVlabs/alpamayo1.5) as an example: (1) a stage 1 ([ref](https://github.com/NVlabs/alpamayo-recipes/tree/main/recipes/alpamayo1_5_sft#training-pipeline)) VLM SFT, (2) a VLM RL post train step ([ref](https://github.com/NVlabs/alpamayo-recipes/tree/main/recipes/alpamayo1_x_rl)), (3) a stage 2 ([ref](https://github.com/NVlabs/alpamayo-recipes/tree/main/recipes/alpamayo1_5_sft#training-pipeline)) action expert SFT, (4) more stages as needed.

With all those said, definitely stay tuned for potential future releases on RL post training recipes for the action expert as well!
